src/main/java/com/puppycrawl/tools/checkstyle/checks/TrailingCommentCheck.java

```diff
@@ -35,7 +35,7 @@
 
 /**
  * <p>
- * The check to ensure that requires that comments be the only thing on a line.
+ * The check to ensure that lines do not end with comment.
```

Comment:
My recommendation is:

The check to ensure that lines with code do not end with comment.

A line that has only a comment should be acceptable. I am not sure if `code` is acceptable as it needs to include `}`/`{` like the those attached to ifs, loops, methods, etc...

Reply:
applied